EIPS/eip-6493.md

```diff
@@ -55,96 +55,55 @@ When representing such a transaction as part of the [execution block header's `t
+When an SSZ transaction is signed, additional information is mixed into the signed hash to uniquely identify the underlying SSZ scheme as well as the operating network.
 
 | Name | SSZ equivalent | Description |
 | - | - | - |
-| `tx_type` | `TransactionType` | Assigned [EIP-2718](./eip-2718) transaction type |
-| `tx_type_fork_version` | [`Version`](https://github.com/ethereum/consensus-specs/blob/67c2f9ee9eb562f7cc02b2ff90d92c56137944e1/specs/phase0/beacon-chain.md#custom-types) | Historic [EIP-2124](./eip-2124.md#specification) `FORK_HASH` of type's introduction |
-| `genesis_hash` | [`Hash32`](https://github.com/ethereum/consensus-specs/blob/67c2f9ee9eb562f7cc02b2ff90d92c56137944e1/specs/phase0/beacon-chain.md#custom-types) | Blockhash of the first block on the chain |
 | `chain_id` | `uint256` | [EIP-155](./eip-155.md) chain ID at time of signature |
+| `tx_type` | `TransactionType` | Assigned [EIP-2718](./eip-2718) transaction type |
 
 The following helper functions compute the [`Domain`](https://github.com/ethereum/consensus-specs/blob/67c2f9ee9eb562f7cc02b2ff90d92c56137944e1/specs/phase0/beacon-chain.md#custom-types) for signing an SSZ transaction.
 
 ```python
-class ExecutionForkData(Container):
-    fork_version: Version
-    genesis_hash: Hash32
+class TransactionDomainData(Container):
+    tx_type: TransactionType
     chain_id: uint256
 
-def compute_execution_fork_data_root(
-    fork_version: Version,
-    genesis_hash: Hash32,
-    chain_id: uint256,
-) -> Root:
-    return ExecutionForkData(
-        fork_version=fork_version,
-        genesis_hash=genesis_hash,
-        chain_id=chain_id,
-    ).hash_tree_root()
-
-def compute_execution_domain(
-    domain_type: DomainType,
-    fork_version: Version,
-    genesis_hash: Hash32,
-    chain_id: uint256,
-) -> Domain:
-    fork_data_root = compute_execution_fork_data_root(fork_version, genesis_hash, chain_id)
-    return Domain(domain_type + fork_data_root[:28])
-
 def compute_transaction_domain(
     tx_type: TransactionType,
-    tx_type_fork_version: Version,
-    genesis_hash: Hash32,
     chain_id: uint256,
 ) -> Domain:
-    domain_type = domain_type_for_transaction_type(tx_type)
-    return compute_execution_domain(domain_type, tx_type_fork_version, genesis_hash, chain_id)
+    return Domain(TransactionDomainData(
+        tx_type=tx_type,
+        chain_id=chain_id,
+    ).hash_tree_root())
 ```
 
 The [`Root`](https://github.com/ethereum/consensus-specs/blob/67c2f9ee9eb562f7cc02b2ff90d92c56137944e1/specs/phase0/beacon-chain.md#custom-types) to sign is computed using [`compute_signing_root`](https://github.com/ethereum/consensus-specs/blob/67c2f9ee9eb562f7cc02b2ff90d92c56137944e1/specs/phase0/beacon-chain.md#compute_signing_root) based on the unsigned transaction's `hash_tree_root` and the additional information about the transaction type.
 
+```python
+def compute_signing_root(ssz_object: SSZObject, domain: Domain) -> Root:
+    return hash_tree_root(SigningData(
```

Comment:
@etan-status Can we clarify the definition of `SigningData`?

Reply:
https://github.com/ethereum/consensus-specs/blob/67c2f9ee9eb562f7cc02b2ff90d92c56137944e1/specs/phase0/beacon-chain.md#signingdata

It's just the pre-existing type. I can copy it here for clarity as well.